### DISCUSSION TITLE
Ask oauth2 provider for more than email?

### DISCUSSION THREAD

**Discussion by holtgrewe** [fastapi-users/fastapi-users#1294] [Q&A] [answered]
First, thank you for the great library.

How would I go about requesting more than just the email from my oauth provider? It this something to request on login or to retrieve somehow afterwards?

Thanks.

**Comment by frankie567** [comment, marked as answer]
Hi!

The OAuth2 process just creates an account with the email retrieved from the provider. However, the user account is then associated to a valid access token you can use to make other API requests and get data (given that you asked for the required scopes). 

Typically, you can do this in the `on_after_register` method.
